tests/sentry/sentry_metrics/test_multiprocess_steps.py

```diff
@@ -29,6 +29,34 @@
 logger = logging.getLogger(__name__)
 
 
+def compare_messages_ignoring_mapping_metadata(actual: Message, expected: Message) -> None:
+    assert actual.offset == expected.offset
+    assert actual.partition == expected.partition
+    assert actual.timestamp == expected.timestamp
```

Comment:
I'm open to better ways of doing this (iterating over `__slots__` or `__dict__`, maybe?)

Reply:
Do you want to compare all the fields or just some?

Some:
```
fields = ["offset", "partition", "timestamp"]
for f in fields:
    assert getattr(actual, f) == getattr(expected, f), f"{f} did not match"
```

Reply:
I'm trying to compare all the fields except for the new ones I added because they're not that relevant to the existing tests (if we want to change the "data source" for these fields, I don't think it affects the e2e functionality that much).

I don't feel very strongly but I find my code more readable/clear than the looping style. I think what I'm asking here is if there's some idiom for: 
1. fetch **all** fields
2. filter that list for one or two I don't care about
3. compare the rest

Reply:
I personally like the explicit comparing you have vs looping (although I know ive done loops for tests in other places). You could also do a contains in other cases, but here the KafkaPayload makes it a little bit annoying to test since you have to deserialize it. 

If we are mocking the indexer though you could add the metadata to the test (if that's what you mean by changing the "data source").